core/unit_test/TestConcepts.hpp

```diff
@@ -59,4 +59,171 @@ static_assert(std::is_same<int, Kokkos::Impl::remove_cvref_t<int &>>{}, "");
+      decltype(std::declval<U const &>().team_scratch(int{}));
+
+  template <class U>
+  using ThreadScracthArchetypeExpr =
+      decltype(std::declval<U const &>().thread_scratch(int{}));
+
+  // collectives
+  template <class U>
+  using TeamBarrierArchetypeExpr =
+      decltype(std::declval<U const &>().team_barrier());
+
+  template <class U>
+  using TeamBroadcastArchetypeExpr =
+      decltype(std::declval<U const &>().team_broadcast(
+          lvalueForMethodsNeedingIt_, int{}));
+
+  template <class U>
+  using TeamBroadcastAcceptClosureArchetypeExpr =
+      decltype(std::declval<U const &>().team_broadcast(
+          TrivialFunctor{}, lvalueForMethodsNeedingIt_, int{}));
+
+  template <class U>
+  using TeamReducedArchetypeExpr =
+      decltype(std::declval<U const &>().team_reduce(
+          std::declval<reduction_to_test_t>()));
+
+  template <class U>
+  using TeamScanArchetypeExpr = decltype(std::declval<U const &>().team_scan(
+      lvalueForMethodsNeedingIt_, ptrForMethodsNeedingIt_));
+
+ public:
+  static constexpr bool value =
+      Kokkos::is_detected_v<ExecutionSpaceArchetypeAlias, T> &&
+      Kokkos::is_detected_v<ScratchMemorySpaceArchetypeAlias, T> &&
+      //
+      Kokkos::is_detected_exact_v<int, TeamRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, TeamSizeArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueRankArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<int, LeagueSizeArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamShmemArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          TeamScratchArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<
+          Kokkos::detected_t<ScratchMemorySpaceArchetypeAlias, T> const &,
+          ThreadScracthArchetypeExpr, T> &&
+      //
+      Kokkos::is_detected_exact_v<void, TeamBarrierArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<void, TeamBroadcastAcceptClosureArchetypeExpr,
+                                  T> &&
+      Kokkos::is_detected_exact_v<void, TeamReducedArchetypeExpr, T> &&
+      Kokkos::is_detected_exact_v<test_value_type, TeamScanArchetypeExpr, T>;
+  constexpr operator bool() const noexcept { return value; }
+};
+
+template <class T>
+inline constexpr bool is_team_handle_complete_trait_check_v =
+    is_team_handle_complete_trait_check<T>::value;
+
+// actual test begins here
+
+/*
+  disabling as follows:
+
+  - OpenMPTARGET: due to this
+    https://github.com/kokkos/kokkos/blob/2d6cbad7e079eb45ae69ac6a59929d9fcf10409a/core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp#L860
+
+    - OpenACC: not supporting teams yet
+   */
+#if not defined KOKKOS_ENABLE_OPENMPTARGET && not defined KOKKOS_ENABLE_OPENACC
```

Comment:
Do not introduce a different style
```suggestion
#if !defined(KOKKOS_ENABLE_OPENMPTARGET) && !defined(KOKKOS_ENABLE_OPENACC)
```